validator/sawtooth_validator/concurrent/atomic.py

```diff
@@ -46,3 +47,80 @@ def inc(self, step=1):
     def dec(self, step=1):
         with self._lock:
             self._value -= step
+
+
+class ConcurrentSet:
+    def __init__(self):
+        self.set = set()
+        self.lock = RLock()
```

Comment:
Though RLock can be used in place of Lock, it doesn't seem necessary here. https://stackoverflow.com/a/22885810/1766739 This stackoverflow answer describes the use of RLock; when calling functions that are synchronized with the same lock. I would also use `self._lock` instead of `self.lock`. The protection we get from the linter that an internal lock doesn't get called externally to the object by accident is important.

Reply:
I'll fix `self._lock`. I am not going to change RLock to Lock in this PR because I was not able to convince myself that the lock didn't need to be re-entrant when I wrote this in the first place and RLock has no performance cost over Lock: https://docs.python.org/3/whatsnew/3.2.html#optimizations "Recursive locks (created with the threading.RLock() API) now benefit from a C implementation which makes them as fast as regular locks, and between 10x and 15x faster than their previous pure Python implementation."

Reply:
Thanks, That was news to me.